### PR TITLE
COMMERCE-5329 Add integration tests

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 1]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
+
+		CPDefinition cpDefinition = cpInstance.getCPDefinition();
+
+		BigDecimal price = BigDecimal.valueOf(100);
+		BigDecimal promoPrice = BigDecimal.valueOf(80);
+		BigDecimal tierPrice = BigDecimal.valueOf(50);
+
+		CommercePriceEntry commercePriceEntry =
+			CommercePriceEntryTestUtil.addCommercePriceEntry(
+				"", cpDefinition.getCProductId(),
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:309]: We should use StringPool.BLANK.

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 2]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
+				cpInstance.getCPInstanceUuid(),
+				commercePriceList.getCommercePriceListId(), price, false, null,
+				null, null, null, true, true);
+
+		CommercePriceEntryTestUtil.addCommerceTierPriceEntry(
+			commercePriceEntry.getCommercePriceEntryId(), "", tierPrice, 5,
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:315]: We should use StringPool.BLANK.

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 3]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
+			false, false, null, null, null, null, true, true);
+
+		CommercePriceEntryTestUtil.addCommercePriceEntry(
+			"", cpDefinition.getCProductId(), cpInstance.getCPInstanceUuid(),
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:319]: We should use StringPool.BLANK.

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 4]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
+		CPInstance cpInstance = CPTestUtil.addCPInstanceFromCatalog(
+			catalog.getGroupId());
+
+		CPDefinition cpDefinition = cpInstance.getCPDefinition();
+
+		BigDecimal price = BigDecimal.valueOf(100);
+		BigDecimal promoPrice = BigDecimal.valueOf(500);
+
+		CommercePriceEntryTestUtil.addCommercePriceEntry(
+			"", cpDefinition.getCProductId(), cpInstance.getCPInstanceUuid(),
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:394]: We should use StringPool.BLANK.

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 5]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
+			commercePriceList.getCommercePriceListId(), price, false, null,
+			null, null, null, true, true);
+
+		CommercePriceEntryTestUtil.addCommercePriceEntry(
+			"", cpDefinition.getCProductId(), cpInstance.getCPInstanceUuid(),
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:399]: We should use StringPool.BLANK.

**Review by dsitu — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 6]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
+		).given(
+			"A catalog with a product a price list with a price entry with " +
+				"the product and a promo on the product"
+		).when(
+			"The price of the product is discovered"
+		).then(
+			"The correct price and the promo is returned "
+		);
+
+		CommerceCatalog catalog =
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:286]: We should name this `commerceCatalog`

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 7]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:373]: Name this `commerceCatalog`.

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 8]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
+			_commerceCatalogLocalService.addCommerceCatalog(
+				RandomTestUtil.randomString(), _commerceCurrency.getCode(),
+				LocaleUtil.US.getDisplayLanguage(), null, _serviceContext);
+
+		CommercePriceList commercePriceList =
+			CommercePriceListTestUtil.addCommercePriceList(
+				catalog.getGroupId(), 0.0);
+
+		CommercePriceList commercePromotion =
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:295]: Maybe we can name this `promotionalCommercePriceList`.

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java [hunk 9]
```diff
@@ -260,6 +269,173 @@ public void testEmptyDiscountInPriceEntryDiscoveryFalse() throws Exception {
...
```
Comment (dsitu) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/pricing/test/CommercePricingTest.java:382]: Maybe we can name this `promotionalCommercePriceList`.

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Review by dsitu — APPROVED** [review]

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:forward

**Pull request closed without merging**